android-auto-app/build.gradle.kts

```diff
@@ -32,6 +32,10 @@ android {
     }
   }
 
+  kotlinOptions {
+    freeCompilerArgs = freeCompilerArgs + "-Xjvm-default=all"
```

Comment:
Am testing this on a bunch of interfaces to see if we catch any issues on the navigation side

https://github.com/mapbox/mapbox-navigation-android/pull/6335#issuecomment-1248690974
^ Here are some details that show that adding the compiler flag may force some projects to add the compiler option. That seems like an issue with the compiler flag 🤔

Reply:
For this reason I'm preferring the `java interface` approach

Reply:
I mean, `jvm-default=all` works well if you are building an `app` but it has as different behavior when building an `sdk`.

specifically, when I added `-Xjvm-default=all` to the `nav-sdk` it broke a downstream build because it is not using `-Xjvm-default=all` that is using kotlin. So a new SDK like android-auto could require downstream to add the flag, but this unpredictability is too concerning IMO.

What I mean, is define the interface with java so that the sdk is not depending on a "kotlin to java" converter.

Reply:
Notice how this pull request requires the addition of the compiler flat in both build.gradle files
````
android-auto-app/build.gradle.kts
extension-androidauto/build.gradle.kts
```

Reply:
This may be due to the fact it is not a released SDK though and it's being compiled. The mobile-metrics may be an issue with how dependencies are being injected. Will test this out with a snapshot.

Reply:
@yunikkk this pull request is updated with the approach I'm thinking we need.

I've moved the jvm-default flag approach over here https://github.com/mapbox/mapbox-maps-android/pull/1674

Reply:
LGTM, I'm fine with the approach.

BTW could you elaborate the issue you've faced adding with the `-Xjvm-default=all` flag? Were you able to fix using snapshot?

Reply:
Yeah I've updated this branch to use the `jvm-default` issue https://github.com/mapbox/mapbox-maps-android/pull/1674/files#r975672871

That way anyone can pull that branch and reproduce the issue

Reply:
https://github.com/mapbox/mapbox-maps-android/pull/1670#discussion_r973246052

This comment describes the issue well. The `-Xjvm-default` compiler flag may do what we want in the future. But right now, it does not generate default interfaces in an SDK. The app developer has to include it.

One big issue for us to not add it, is kotlin will fail to compile if the app does not include it - totally unexpected behavior in my opinion because I would only expect java to fail to compile.

Reply:
> But right now, it does not generate default interfaces in an SDK.

Unexpected indeed, thanks for the investigation